pallets/runtime/tests/src/contract_test.rs

```diff
@@ -694,3 +706,45 @@ fn check_meta_url_functionality() {
             ));
         });
 }
+
+#[test]
+fn check_put_code_flag() {
+    let gc = AccountKeyring::Bob.public();
+    let user = AccountKeyring::Charlie.public();
+
+    ExtBuilder::default()
+        .governance_committee(vec![gc])
+        .regular_users(vec![user])
+        .build()
+        .execute_with(|| check_put_code_flag_ext(gc, user))
+}
+
+fn check_put_code_flag_ext(gc_member: Public, user: Public) {
+    let (wasm, _) = compile_module::<TestStorage>("flipper").unwrap();
+    assert!(user != gc_member);
+
+    let put_code = |acc: Public| -> DispatchResult {
+        WrapperContracts::put_code(
+            Origin::signed(acc),
+            TemplateMetadata::default(),
+            0u128,
+            wasm.clone(),
+        )
+    };
+    let enable_put_code = |acc: Public| -> DispatchResult {
+        WrapperContracts::set_put_code_flag(Origin::signed(acc), true)
+    };
+
+    // Flag is disable, so `put_code` should fail.
+    assert_err!(put_code(user), WrapperContractsError::PutCodeIsNotAllowed);
+
+    // Non GC member cannot update the flag.
+    assert_err!(
```

Comment:
```suggestion
    assert_noop!(put_code(user), WrapperContractsError::PutCodeIsNotAllowed);

    // Non GC member cannot update the flag.
    assert_noop!(
```
Let's always use `_noop` unless a state change on failure is possible, in which case that should be motivated with a comment.